### DISCUSSION TITLE
setting the array to empty `subscribe` fails to subscribe.

### DISCUSSION THREAD

**Discussion by ayangweb** [pmndrs/valtio#999] [Q&A]
```ts
export const store = proxy({
	arr: [1, 2, 3],
});

subscribe(store.arr, (value) => {
	// No output when array is empty
	console.log("value", value);
});

store.arr = [];
```

1. setting the array to empty `subscribe` fails to subscribe.
2. store is the persistent store `[]`, which is still the default value `[1, 2, 3]` when the project starts.

**Comment by dai-shi** [comment]
`store.arr = [];` means you are mutating `store` not `store.arr`. So, you need to subscribe to `store`.

```js
subscribe(store, () => {
	console.log("value", store.arr);
});
```

**Comment by ayangweb** [comment]
Ok, `subscribe` does work, so what should I do about the second question?

**Comment by dai-shi** [comment]
> 2\. store is the persistent store `[]`, which is still the default value `[1, 2, 3]` when the project starts.

Can you elaborate the question?

**Comment by ayangweb** [comment]
Well, the problem is also related to the fact that `subscribe(store.arr)` is not working!

Is there any other simple way to listen for changes when arr is empty? `subscribe(store)` is a bit complicated.

**Reply by dai-shi** [reply]
I don't the exact problem, but how about `subscribeKey(store, 'arr', () => {})`?

It's just a simple wrapper.
https://github.com/pmndrs/valtio/blob/e21edb3828cc8e6b0e54c0b258ae4657e7268551/src/vanilla/utils/subscribeKey.ts#L14-L31

**Comment by ayangweb** [comment]
`subscribeKey` I've tested it, can't listen for changes to objects and arrays, only normal types of data!

**Comment by ayangweb** [comment]
Wait, `subscribeKey` seems to be working again, I'll get back to you when I verify it again!

**Reply by dai-shi** [reply]
even if it doesn't work for your case, you can do something similar and adjust for your case. underneath it's just `subscribe`.

**Comment by ayangweb** [comment]
The `subscribeKey` is working, but in Tauri development, since I'm using multiple windows, it's not possible for other windows to listen for changes via `subscribeKey`. Thanks for the heads up, I'll look into this further, thanks!
